beamer/tests/contracts/test_request_manager_fees.py

```diff
@@ -4,120 +4,164 @@
     claim_id = claim_tx.return_value
 
+    # Update fees, which should not have any effect on the fee amounts that
+    # were computed when the request was made.
+    request_manager.updateFeeData(17e9, 145_000, 21_000, {"from": deployer})
+
     # Timetravel after claim period
     chain.mine(timedelta=claim_period)
 
     # Even if the requester calls withdraw, the funds go to the claimer
     withdraw_tx = request_manager.withdraw(claim_id, {"from": requester})
     assert "ClaimWithdrawn" in withdraw_tx.events
+    assert request_manager.collectedProtocolFees(token) == protocol_fee
+    assert token.balanceOf(request_manager) == protocol_fee
+    assert token.balanceOf(claimer) == transfer_amount + lp_fee
+    assert web3.eth.get_balance(claimer.address) == old_claimer_eth + reimbursement_fee
+
 
-    assert request_manager.collectedBeamerFees() == beamer_fee
-    assert request_manager.requests(request_id)[9] == reimbursement_fee + lp_service_fee
-    assert request_manager.requests(request_id)[10] == beamer_fee
+def test_protocol_fee_is_zero(request_manager):
+    # For the time being, the protocol fee percentage should be zero.
+    assert request_manager.protocolFeeTokenPPM() == 0
+    assert request_manager.protocolFeeToken(23_000_000) == 0
 
 
-def test_beamer_service_fee_withdrawable_by_owner(
+def test_protocol_fee_withdrawable_by_owner(
     deployer, request_manager, token, claim_stake, claim_period
 ):
     owner = deployer
     requester, claimer = alloc_accounts(2)
-    beamer_fee = request_manager.beamerServiceFee()
-    request_id = make_request(request_manager, token, requester, requester, 23, zero_fees=False)
+    amount = 23_000_000
+    request_id = make_request(
+        request_manager, token, requester, requester, amount, fee_data=_NONZERO_FEE_DATA
+    )
+    protocol_fee = request_manager.requests(request_id)[RM_FIELD_PROTOCOL_FEE]
 
     with brownie.reverts("Ownable: caller is not the owner"):
-        request_manager.withdrawbeamerFees({"from": requester})
+        request_manager.withdrawProtocolFeeToken(token, requester, {"from": requester})
 
-    assert request_manager.collectedBeamerFees() == 0
-    with brownie.reverts("Zero fees available"):
-        request_manager.withdrawbeamerFees({"from": owner})
+    with brownie.reverts("Protocol fee is zero"):
+        request_manager.withdrawProtocolFeeToken(token, owner, {"from": owner})
 
     claim_tx = request_manager.claimRequest(request_id, 0, {"from": claimer, "value": claim_stake})
     claim_id = claim_tx.return_value
 
     chain.mine(timedelta=claim_period)
 
-    assert request_manager.collectedBeamerFees() == 0
-    with brownie.reverts("Zero fees available"):
-        request_manager.withdrawbeamerFees({"from": owner})
+    with brownie.reverts("Protocol fee is zero"):
+        request_manager.withdrawProtocolFeeToken(token, owner, {"from": owner})
 
     request_manager.withdraw(claim_id, {"from": requester})
```

Comment:
It would be good to change this to `{"from": claimer}`.

Reply:
Done.